react/src/components/AgentDetailModal.tsx

```diff
@@ -94,10 +94,10 @@ const AgentDetailModal: React.FC<AgentDetailModalProps> = ({
                 <BAIProgressWithLabel
                   percent={
                     ((iSizeToSize(_.toString(agent?.mem_cur_bytes), 'g')
-                      ?.number ?? 0) /
-                      (iSizeToSize(parsedAvailableSlots?.mem, 'g')?.number ??
+                      ?.number || 0) /
+                      (iSizeToSize(parsedAvailableSlots?.mem, 'g')?.number ||
                         0)) *
-                      100 ?? 0
+                    100
```

Comment:
```suggestion
                    (_.toNumber(
                      iSizeToSize(_.toString(agent?.mem_cur_bytes), 'g')
                        ?.number,
                    ) /
                      _.toNumber(
                        iSizeToSize(parsedAvailableSlots?.mem, 'g')?.number,
                      )) *
                      100 || 0
```

To further improve, instead of using `|| 0` to handle NaN, we can modify `BAIProgressWithLabel` to display the percent as disabled or distinguish it in some way when it is NaN.